src/chatClient.cpp

```diff
@@ -1467,11 +1458,55 @@ PeerChatRoom::~PeerChatRoom()
         chatd->leave(mChatid);
 }
 
+void PeerChatRoom::initContact(const uint64_t& peer)
+{
+    mContact = parent.client.contactList->contactFromUserId(peer);
+    mEmail = mContact ? mContact->email() : "Inactive account";
```

Comment:
You can prevent duplicating the email property in the contact and 1on1 room by selectively returning here mContact->mEmail or "Deleted user"

Reply:
I think the memory used by `mEmail` in the `ChatRoom` object is worthy, since the `Contact` may be deleted and also to speed up the access to the email.